integrations/oci/metrics/metrics/src/main/java/io/helidon/integrations/oci/metrics/OciMetricsSupport.java

```diff
@@ -0,0 +1,527 @@
+            LOGGER.fine("Shutting down OCI Metrics agent");
+            scheduledExecutorService.shutdownNow();
+        }
+    }
+
+    private void startExecutor() {
+        scheduledExecutorService = Executors.newScheduledThreadPool(1);
+        scheduledExecutorService.scheduleAtFixedRate(this::pushMetrics, initialDelay, delay, schedulingTimeUnit);
+    }
+
+    private void pushMetrics() {
+        List<MetricDataDetails> allMetricDataDetails = ociMetricsData.getMetricDataDetails();
+        LOGGER.finest(String.format("Processing %d metrics", allMetricDataDetails.size()));
+
+        if (allMetricDataDetails.size() > 0) {
+            while (true) {
+                if (allMetricDataDetails.size() > batchSize) {
+                    postBatch(allMetricDataDetails.subList(0, batchSize));
+                    // discard metrics that had been posted
+                    allMetricDataDetails.subList(0, batchSize).clear();
+                    if (batchDelay > 0L) {
+                        try {
+                            schedulingTimeUnit.sleep(batchDelay);
+                        } catch (InterruptedException ignore) {
+                        }
+                    }
+                } else {
+                    postBatch(allMetricDataDetails);
+                    break;
+                }
+            }
+        }
+    }
+
+    private void postBatch(List<MetricDataDetails> metricDataDetailsList) {
+        PostMetricDataDetails postMetricDataDetails = PostMetricDataDetails.builder()
+                .metricData(metricDataDetailsList)
+                .build();
+
+        PostMetricDataRequest postMetricDataRequest = PostMetricDataRequest.builder()
+                .postMetricDataDetails(postMetricDataDetails)
+                .build();
+
+        LOGGER.finest(String.format("Pushing %d metrics to OCI", metricDataDetailsList.size()));
+        if (LOGGER.isLoggable(Level.FINEST)) {
+            metricDataDetailsList
+                    .stream()
+                    .forEach(m -> {
+                        LOGGER.finest(String.format(
+                                "Metric details: name=%s, namespace=%s, dimensions=%s, "
+                                        + "datapoints.timestamp=%s, datapoints.value=%f, metadata=%s",
+                                m.getName(),
+                                m.getNamespace(),
+                                m.getDimensions(),
+                                m.getDatapoints().get(0).getTimestamp(),
+                                m.getDatapoints().get(0).getValue(),
+                                m.getMetadata()));
+                    });
+        }
+        String originalMonitoringEndpoint = this.monitoringClient.getEndpoint();
+        try {
+            // Use the ingestion endpoint for posting
+            this.monitoringClient.setEndpoint(
+                    monitoringClient.getEndpoint().replaceFirst("telemetry\\.", "telemetry-ingestion."));
```

Comment:
Are the paths always these particular strings? Is there any time they might be different and, therefore, need to be configurable on the Helidon integration side?

Reply:
Yeah, that is a fix value: https://docs.oracle.com/en-us/iaas/api/#/en/monitoring/20180401/. We use to do the same magically via the filter of the [OCI client's sdk](https://github.com/helidon-io/helidon/blob/helidon-2.x/integrations/oci/sdk/cdi/src/main/java/io/helidon/integrations/oci/sdk/cdi/OciExtension.java#L1378), but that will work anymore in OCI Java SDK v3.